File: r2dbc-db/src/main/java/com/example/r2dbc/repository/PersonRepository.java
package com.example.r2dbc.repository;


import com.example.r2dbc.model.PersonReactive;
import org.springframework.data.r2dbc.repository.query.Query;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

@Repository
public interface PersonRepository extends ReactiveCrudRepository<PersonReactive,Long> {

    @Query("select id, name from public.person e where e.name = $1")
    Flux<PersonReactive> findByName(String name) ;

    @Query("select id, name, age from public.person where deleted_at is null")
    Flux<PersonReactive> findPeople() ;

}

File: kotlin-r2dbc/src/main/kotlin/r2dbc/kotlin/repository/PersonRepository.kt
package r2dbc.kotlin.repository

import kotlinx.coroutines.FlowPreview
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.reactive.awaitFirstOrNull
import kotlinx.coroutines.reactive.flow.asFlow
import org.springframework.data.r2dbc.function.DatabaseClient
import org.springframework.stereotype.Repository
import r2dbc.kotlin.model.Person

@Repository
class PersonRepository(private val client: DatabaseClient) {

    suspend fun getById(id: Long): Person? = client.execute()
            .sql("select id, name, age from person where id=$1")
            .bind(0, id)
            .`as`(Person::class.java)
            .fetch()
            .one()
            .awaitFirstOrNull()

    @FlowPreview
    suspend fun findAll(): Flow<Person> = client.execute()
            .sql("select id, name, age from person where deleted_at is null")
            .`as`(Person::class.java)
            .fetch()
            .all()
            .asFlow()
}